### PR TITLE
Using Mammals and Reptiles classes

### PR DESCRIPTION
**Pull request by Kamen06**
Removing property "group". Adding two abstract classes - Mammal and Reptile. Removing getGroup() method.

### CODE REVIEW

**Review by teodora1203 — CHANGES_REQUESTED** [review]

File: src/hierarchy/Reptile.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import Animal from "./Animal";
+
+export default class Reptile extends Animal {
```
[the remark this answers is not in the document]
Comment (Kamen06) [reply src/hierarchy/Reptile.ts:3]: Must have forgotten. Will do!

File: src/hierarchy/Reptile.ts [hunk 2]
```diff
@@ -1,6 +1,11 @@
+import { ICanSwim } from "../interfaces/ICanSwim";
 import Animal from "./Animal";
 
-export default class Reptile extends Animal {
+export default abstract class Reptile extends Animal implements ICanSwim{
+
+    swim() {
+        console.log(`${this.name} is swiming`);
```
Comment (rkoychev) [inline src/hierarchy/Reptile.ts:7]: swimming should be with double "m"

File: src/animals/Crocodile.ts [hunk 3]
```diff
@@ -1,31 +1,21 @@
...
-    this.group = "Reptile";
-  }
-  swim() {
-    console.log(`${this.name} is swimming`);
-  }
-  walk() {
-    console.log(`${this.name} is walking`);
   }
+  
+
```
Comment (rkoychev) [inline src/animals/Crocodile.ts:18]: extra empty line

File: src/hierarchy/Mammal.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { ICanJump } from "../interfaces/ICanJump";
+import Animal from "./Animal";
+
+export default abstract class Mammal extends Animal implements ICanJump {
+    jump(): void {
+        console.log(`${this.name} is jumping`);
+    }
+
```
Comment (rkoychev) [inline src/hierarchy/Mammal.ts:8]: a lot of extra lines

**Review by rkoychev — APPROVED** [review]
a few minor comments that can be addressed in a future  pr.
you can merge this one

**Review by teodora1203 — APPROVED** [review]

**Pull request merged**